Helpers/get_results_from_criteria_and_data.py

```diff
@@ -0,0 +1,130 @@
+                    desired_checkin=desired_checkin,
+                    desired_checkout=desired_checkout,
+                    house=house,
+                )
+            results_list.append(data_check)
+
+        return results_list
+
+    @staticmethod
+    def update_data_check_when_both_checkin_checkout_provided(
+            desired_checkin: date,
+            desired_checkout: date,
+            house: House
+    ) -> DatesCheck:
+        """
+        Updates the required parameters of the DatesCheck object when both checkin and checkout are provided.
+        """
+        is_available = False
+        for date_ranges in house.available_ranges:
+            if (
+                    desired_checkin >= date_ranges.range_start and
+                    desired_checkout <= date_ranges.range_end and
+                    int((desired_checkout - desired_checkin).days) >= house.min_stay_nights
+            ):
+                is_available = True
+                break
+
+        return DatesCheck(
+            listing_id=house.listing_id,
+            is_available=is_available,
+            checkin_date=desired_checkin,
+            checkout_date=desired_checkout
+        )
+
+    @staticmethod
+    def update_data_check_when_only_checkin_provided(
+            desired_checkin: date,
+            house: House
+    ) -> DatesCheck:
+        """
+        Updates the required parameters of the DatesCheck Object when only checkin is provided
+        """
+        checkin_date = desired_checkin
+        checkout_date = None
+        is_available = False
+        for date_ranges in house.available_ranges:
+            if (
+                    desired_checkin >= date_ranges.range_start and
+                    desired_checkin + timedelta(days=house.min_stay_nights) <= date_ranges.range_end
+            ):
+                checkout_date = desired_checkin + timedelta(days=house.min_stay_nights)
+                is_available = True
+                break
+            elif desired_checkin > date_ranges.range_start:
+                continue
+            else:
+                is_available = True
+                checkin_date = date_ranges.range_start
+                checkout_date = date_ranges.range_start + timedelta(days=house.min_stay_nights)
+                break
+
+        return DatesCheck(
+            listing_id=house.listing_id,
+            is_available=is_available,
+            checkin_date=checkin_date,
+            checkout_date=checkout_date
+
+        )
+
+    @staticmethod
+    def update_data_check_when_null_checkin_checkout(
+            house: House
+    ) -> DatesCheck:
+        """
+        Update and create the DatesCheck object when neighter checkin nor checkout is provided
+        #Assumption : There is a always a house in the data with available dates
+        """
+        if date.today() > house.available_ranges[0].range_start:
```

Comment:
And check that `min_stay_nights` requirements is met.
Add that the assumption is that only current/future available ranges are stored for each listing

Reply:
Not needed as discussed under the current assumptions